api/v3/FinancialTrxn.php

```diff
@@ -39,7 +39,11 @@
  * @return array
  */
 function civicrm_api3_financial_trxn_create($params) {
-  return _civicrm_api3_basic_create(_civicrm_api3_get_BAO(__FUNCTION__), $params);
+  if (empty($params['id']) && empty($params['contribution_id']) && empty($params['entity_id'])) {
```

Comment:
I think it can do a combo of AND + OR ie

array('x', 'y') is x +y 

array (('x', 'y') is x OR y

array (('x', array('y', 'z') ) is x AND y or Z